docs/architecture/adr/adr-006-in-cluster-db.md

```diff
@@ -0,0 +1,170 @@
+# ADR 006: In-cluster Database
+
+## Context
+
+We have known, since summer 2024, that Microsoft had decided to sunset Azure
+database for MariaDB. Their recommendation has been to migrate to their Flexible
+MySql database which they have provided pretty well documented. Back then we had
+already experienced database downtime, which prompted us to start a discussion
+about wether to move the databse to Azure Flexible MySql server or to an
+in-cluster database. This conversation has been running for the better part of
+6-8 months.
+
+The main points we kept debating was:
+
+- Can we operate an in-cluster database on par or better than Azure?
+- Are we able to deal with problems that might arise on the databases with out
+Azures Technical Support?
+- If we self host the databse, are we then able to optimize the performance?
+- Can we guarantee a better or on par uptime to Azure's managed database if we
+selfhost the database?
+- Would we get better support if we moved to the suggested managed database?
+- Would the performance be better on the database suggested by Azure?
+- Would downtime be minimized if we moved to the sugggested manged database?
```

Comment:
I don't think a list of question works well in a decision document. Stick to the answers instead of the questions.

Maybe these questions can be turned into headings for sections you go through to discuss the issue?